Library/Homebrew/formula_installer.rb

```diff
@@ -1076,10 +1076,10 @@ def post_install
       #{HOMEBREW_LIBRARY_PATH}/postinstall.rb
     ]
 
-    args << if formula.local_bottle_path.present?
-      formula.prefix/".brew/#{formula.name}.rb"
-    else
+    args << if formula.path.exist?
       formula.path
+    else
+      formula.prefix/".brew/#{formula.name}.rb"
```

Comment:
Sorry, I'm not sure this is what you want because I do think the `.brew` one should take priority over the one from the tap when e.g. installing from local bottles (and perhaps other cases).

Reply:
Ugh, yes. I think I misunderstood based on your previous suggestion, that would use the prefix formula only if the tap formula didn't exist.

Reply:
I might be overthinking this, but it seems to be more complicated than what I initially thought, so here's what I think:

* Source install, `HEAD` install, and modified-formulae install/postinstall:
  * Use the formula from the Tap since the Tap must be installed for these methods.
* Normal (no changes) install/postinstall and local bottle install/postinstall:
  * Use the formula from the prefix if it is readable.
  * Else use the formula from the Tap (to account for cases like https://github.com/Homebrew/brew/pull/11468#issuecomment-860627703). Such formulae _must_ require the Tap to be installed, unless we make changes to `brew` such that the extra/library files are accessible.

There could be problems with using `brew postinstall` when:
* The formula from the prefix isn't readable but the installed version is older than the Tap version.
* The formula exists in the Tap (which is installed) but an older version is installed.

However, I think that's too much to handle and the user should be asked to upgrade or extract the older version.

This works out to:
```ruby
if formula.local_bottle_path.present? || !formula.tap.exist?
  begin
    prefix_formula if Formulary.factory(prefix_formula) # prefix_formula is readable
  rescue FormulaUnreadableError
    formula.path # prefix_formula is unreadable
  end
else
  formula.path
end
```

What do you think about this? (And apologies for the long comment 😅.)

Reply:
> However, I think that's too much to handle and the user should be asked to upgrade or extract the older version.

Agreed.